src/darwin/Framework/CHIP/CHIPManualSetupPayloadParser.mm

```diff
@@ -31,6 +33,10 @@ - (id)initWithDecimalStringRepresentation:(NSString *)decimalStringRepresentatio
     if (self = [super init]) {
         _decimalStringRepresentation = decimalStringRepresentation;
         _chipManualSetupPayloadParser = new chip::ManualSetupPayloadParser(std::string([decimalStringRepresentation UTF8String]));
+        if (CHIP_NO_ERROR != chip::Platform::MemoryInit()) {
+            CHIP_LOG_ERROR("Error: couldn't initialize platform memory");
+            delete _chipManualSetupPayloadParser;
```

Comment:
should this be set to null as well?

Maybe create the object only if memory init succeeds, instead of creating always and then deleting if memory allocation fails.

I imagine we want memory init first anway, incase ManualSetupPayloadParser somehow needs memory allocation in the future. I think we should not call any CHIP api before memoryinit.

Reply:
I agree, I'll fix it. I followed "convention" I noticed elsewhere.